### PR TITLE
modernization

### PR DESCRIPTION
**Pull request by haoliangyu**
Update dependency and remove git module. All tests passed.

### CODE REVIEW

**Comment by SBats** [comment]
Hi @haoliangyu , thank you for this more than needed contribution !

If I may, I would like to suggest a few things according to our new plugin structure before we can merge.
- remove Makefile and use package.json scripting for simple tasks like testing (and the clean method is native in npm) 
- put the leaflet.almostover.js file in an src folder and change the main property in the package.json accordingly
- put leaflet and leaflet-geometryutil as a peerDependencies instead of direct dependencies. As long as we don't integrate them in the final file, those should be in the environment

If you're short on time or don't know how to do some of those things please let me know. I will complete your work or help you. It's already nice to contribute :)

**Comment by haoliangyu** [comment]
@SBats Please check the new update. I am not sure if it meets all requirements. If not, please add stuff on it directly.

File: package.json [hunk 1]
```diff
@@ -16,9 +16,9 @@
     },
     "license": "MIT",
     "scripts": {
-        "test": "make test"
+        "test": "./node_modules/mocha-phantomjs/bin/mocha-phantomjs test/index.html"
```
Comment (SBats) [inline package.json]: No need for the `./node_modules/mocha-phantomjs/bin/` part. It's one of the good thing with doing node script, it automatically search for binaries in the node_modules folder using node resolution pattern.
So basically : `mocha-phantomjs test/index.html`

File: package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
         "Leaflet",
         "GIS"
     ],
-    "main": "leaflet.almostover.js",
+    "main": "./src/leaflet.almostover.js",
```
Comment (SBats) [inline package.json]: no need for `./` at the beginning

**Comment by SBats** [comment]
Wonderful ! Thank you for all the fixes !

**Pull request merged**